app/models/concerns/foreman/thread_session.rb

```diff
@@ -73,6 +73,7 @@ def current=(o)
             Rails.logger.info "Current user: #{o.login} (#{o.admin? ? 'administrator' : 'regular user'})"
           end
           Rails.logger.debug "Setting current user thread-local variable to " + (o.is_a?(User) ? o.login : 'nil')
+          ::Logging.mdc['user_login'] = o.try(:login)
```

Comment:
How does this know which logger to send this information to? Does it default to `app` logger unless otherwise specified? Can you specify which logger? e.g. if I wanted to do this with `katello` logger?

Reply:
There are two diagnostic contexts: message (mdc) and nested (ndc). The former one can be seen as "global" - fields in here gets logged along every single message. If you want only to attach fields to particular message, then use ndc. There is a helper `with_fields` described in the text of this PR.

Now what you are asking for is something in the middle, I don't have code for that. Can you give examples of fields which should go only into specific logger?